src/experiment/metrics/basicmetrics.jl

```diff
@@ -52,9 +51,8 @@ function (metrics::BasicMetrics{DontTakeObjective, <:BasicHeuristic})(model::CPM
     metrics.nbEpisodes+=1
     push!(metrics.nodeVisited,copy(model.statistics.nodevisitedpersolution))
     push!(metrics.meanNodeVisitedUntilEnd,model.statistics.numberOfNodes)
-    if ! isempty(model.statistics.nodevisitedpersolution)    #infeasible case
-        push!(metrics.meanNodeVisitedUntilfirstSolFound,model.statistics.nodevisitedpersolution[1])
-    end    
+    index = findall(!isnothing, model.statistics.solutions)  #return the list of index of real solution in model.statistics.solutions
+    push!(metrics.meanNodeVisitedUntilfirstSolFound, !isempty(index) ? model.statistics.nodevisitedpersolution[index[1]] : nothing)
```

Comment:
In the general case, we can have more than one solution. 

> If so, I don't understand why we only consider the first solution in the push! right after.
This is because we are looking for the number of nodes until the first solution is found : 
`    push!(metrics.meanNodeVisitedUntilfirstSolFound, !isempty(index) ? model.statistics.nodevisitedpersolution[index[1]] : nothing)
`

> If not, it might be better to use findfirst to make the code clearer and a bit faster (and also update the comment)
Your point is correct, `findfirst` will be the best practice.